### ISSUE TITLE
clicking the reply to a post link when reading as unauthenticated reader leads to a key error

### ISSUE THREAD

**Issue by jlin** [bigboringsystem/bigboringsystem#142]
Ideally, it would redirect you to the login page to fill in your phone number.

**Comment by coleGillespie** [comment]
or maybe a throw a 403 @jlin

**Comment by coleGillespie** [comment]
this logic is bring thrown from the posts.js controller starting at line 342. uid is undefined when a user is not logged in.
```javascript
  var uid = request.session.get('uid');

  profile.getByUID(uid, function (err, user) {
    if (err) {
      return reply(Boom.wrap(err, 404));
    }
```

**Comment by coleGillespie** [comment]
another way to fix this is to only show the link for reply IF there is a valid uid in the session

**Comment by jasonrhodes** [comment]
+1 to hiding the reply link for u authenticated users

**Comment by coleGillespie** [comment]
@jasonrhodes also maybe we should remove all that FE js and make the reply button all server side /cc @ednapiranha

**Comment by coleGillespie** [comment]
@jasonrhodes fixed, pull request needs approval though

**Comment by coleGillespie** [comment]
this can be closed @ednapiranha #163
